### ISSUE TITLE
Use Filesystem disks within Glide

### ISSUE THREAD

**Issue by jasonvarga** [statamic/ideas#341]
Issue based on https://github.com/statamic/cms/pull/1892

It would be great to allow the Glide cache to use a Flysystem disk when using the "cached" mode.
This would allow the cached images to be served directly from S3, for example.

It's a bit pointless in the regular non-cached mode since it has to go through the controller anyway, but that's irrelevant.

The PR was appreciated but it should be doable in a non-breaking way.

**Comment by GioChocolateBro** [comment]
Was just looking at this, would be great!

I think adding another optional key to the config disk => null would already solved the backwards compatibility.
The cachePath function could just check which to use. ServerFactory of glide already checks if the passed param should be a local path or is a disk.

If we're also able to set a base url ourselves we can get cached image from a CDN/Cloudfront instead of S3.

**Comment by Christophvh** [comment]
Would love this, but the thing I find very weird about this is why are presets or derived images considered cache? Let's say we have 10 variations of a big image. Those variations might also be pretty big. Putting that inside a cache driver seems very strange they just belong in the same place as the original images.

If we look at the very popular library laravel-medialibrary this just stores variations inside a sub-folder on the same location. 

Using a CDN/Cloudfront for the caching part makes more sense to me.

**Comment by jasonvarga** [comment]
Any generated images are considered cache.

**Comment by Christophvh** [comment]
> Any generated images are considered cache.

yeah exactly my point. in our case 90% of storage space is used by generated images/variations and those are just on the disk so s3 doesn't really do anything.

**Comment by jasonvarga** [comment]
Right. That's the point of this feature request - to be able to put the cache on s3, etc.
